ams/server/src/main/java/com/netease/arctic/server/table/executor/SnapshotsExpiringExecutor.java

```diff
@@ -80,7 +47,10 @@ protected long getNextExecutingTime(TableRuntime tableRuntime) {
 
   @Override
   protected boolean enabled(TableRuntime tableRuntime) {
-    return tableRuntime.getTableConfiguration().isExpireSnapshotEnabled();
+    return tableRuntime.getFormat() == TableFormat.ICEBERG &&
```

Comment:
Does all the Paimon format tables not require services provided by TableExecutor? If so, can we move the filter logic at a higher level?

Reply:
Executor is a fundamental mechanism, and it's difficult to guarantee that Paimon won't use it in the future